### PR TITLE
Added methods

### PR DESCRIPTION
**Pull request by SeanWiz**
Alright I know the flow now let's do this.

### CODE REVIEW

File: src/math/vec.hpp [hunk 1]
```diff
@@ -147,6 +167,31 @@ struct Vec3 {
...
+        T z = pow(this->m_values[2], 2);
+
+        return sqrt(x + y + z);
+    }
+
+    Vec3 normalized() {
+        T x = this->m_values[0] * abs(this->m_values[0]);
+        T y = this->m_values[1] * abs(this->m_values[1]);
+        T z = this->m_values[2] * abs(this->m_values[2]);
+        Vec3 n(x, y, z);
```
Comment (ianw3214) [inline src/math/vec.hpp]: This should be replaced with 
```c++
return Vec3(x, y, z);
```

**Comment by SeanWiz** [comment]
Made the changes to both Vec3 and Vec2.

File: src/math/vec.hpp [hunk 2]
```diff
@@ -56,6 +56,25 @@ struct Vec2 {
...
+
+        return x + y;
+    }
+    T norm() {
+        T x = pow(this->m_values[0], 2);
+        T y = pow(this->m_values[1], 2);
+
+        return sqrt(x + y);
+    }
+    Vec2 normalized() {
```
Comment (darren-moore) [inline src/math/vec.hpp]: I think this function can be something like
```cpp
Vec2 normalized(){
    return *this/norm();
}
```

File: src/math/vec.hpp [hunk 3]
```diff
@@ -147,6 +166,30 @@ struct Vec3 {
...
+
+    T norm() {
+        T x = pow(this->m_values[0], 2);
+        T y = pow(this->m_values[1], 2);
+        T z = pow(this->m_values[2], 2);
+
+        return sqrt(x + y + z);
+    }
+
+    Vec3 normalized() {
```
Comment (darren-moore) [inline src/math/vec.hpp]: See my `Vec2` comment

File: src/math/vec.hpp (hunk 2 above)
Comment (SeanWiz) [reply src/math/vec.hpp]: From what I gathered the norm of a vector and a normalized vector are two different things. The former is the magnitude of the vector and the latter is a vector with a magnitude of 1. As for why they sound the same *shrug*

File: src/math/vec.hpp (hunk 3 above)
Comment (SeanWiz) [reply src/math/vec.hpp]: See my `Vec2` reply

File: src/math/vec.hpp (hunk 2 above)
Comment (darren-moore) [reply src/math/vec.hpp]: That's exactly right. Let me know if this makes sense. I've normed the "normalized" vector, demonstrating its magnitude is 1.
image

Comment (SeanWiz) [reply src/math/vec.hpp]: Oh wait now I understand what you meant. For some reason my brain didn't recognize the "/" in your code as a division sign. Yeah putting it that way makes a lot of sense (and I guess explains the linked root word) so I'll make that fix.

**Pull request merged**